src/api.py

```diff
@@ -23,97 +21,136 @@ def home():
-                    payload = {'error_message':'Invalid blob_type or search field'}
-                # no search args beyond blob_type - return all entries for blob_type
-                elif not user_search:
+                    api_resp = Response(status_code=400,error_message='Must provide blob_type',body=None)
+                    return json.dumps(api_resp.dict())
+                blob_type = user_search['blob_type']
+                if not f.validate_search_fields(user_search):
+                    api_resp= Response(status_code= 400,error_message= 'KeyError: invalid blob_type or search field', body=None)
+                    return json.dumps(api_resp.dict())
+                # blob_type only - return all entries for blob_type
+                elif len(user_search) == 1:
                     matches = f.all_entries(blob_type, cur)
-                    status_code = 200
-                    payload = matches 
+                    api_resp = Response(status_code= 200,error_message= None,body= matches)
+                    return json.dumps(api_resp.dict())
                 else:
                     matches:dict = f.search_metadata(blob_type,user_search,cur)
-                    status_code = 200
-                    payload = matches
+                    api_resp = Response(status_code= 200,error_message= None,body= matches)
+                    return json.dumps(api_resp.dict())
+
             elif request.method == 'POST':
-                #extract info from POST
-                new_blob_info:BlobInfo = BlobInfo.parse_obj(request.get_json()) 
-                blob_type = new_blob_info.blob_type
-                if not blob_type in blob_types.keys():
-                    status_code = 400 
-                    payload = f'{blob_type} table does not exist'
-                blob_b64s = new_blob_info.blob_b64s
-                # turn metadata into instance of specified metadata_class to enforce type hints
-                metadata_inst = blob_types[blob_type].parse_obj(new_blob_info.metadata)
-                # add blob to db 
-                blob_id = f.add_blob(blob_b64s, cur)
-                # create dict with new_entry metadata including blob_id
-                metadata_inst.blob_id = blob_id 
-                metadata_dict = metadata_inst.dict()
-                # add metadata entry to db
-                entry_id = f.add_entry(blob_type, metadata_dict, cur)
-                status_code = 200
-                payload = {'entry_id':entry_id}
+                try:
+                    #extract info from POST
+                    new_blob_info = BlobPostData.parse_obj(request.get_json()) 
+                    blob_type = new_blob_info.metadata['blob_type']
+                    if not blob_type in blob_types.keys():
+                        api_resp = Response(status_code=400,error_message= f'{blob_type} blob_type does not exist', body=None)
+                        return json.dumps(api_resp.dict())
```

Comment:
can you add a method to `Response` called `to_json` that will do this operation for you, so that you have a simpler call?

Reply:
used inbuilt method for pydantic classes model.json()